beacon-chain/sync/rpc.go

```diff
@@ -0,0 +1,53 @@
+package sync
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	libp2pcore "github.com/libp2p/go-libp2p-core"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/prysmaticlabs/prysm/shared/roughtime"
+)
+
+// Time to first byte timeout. The maximum time to wait for first byte of
+// request response (time-to-first-byte). The client is expected to give up if
+// they don't receive the first byte within 5 seconds.
+var ttfbTimeout = 5 * time.Second
+
+// rpcHandler is responsible for handling and responding to any incoming message.
+// This method may return an error to internal monitoring, but the error will
+// not be relayed to the peer.
+type rpcHandler func(context.Context, proto.Message, libp2pcore.Stream) error
+
+// TODO(3147): Delete after all handlers implemented.
+func notImplementedRPCHandler(_ context.Context, _ proto.Message, _ libp2pcore.Stream) error {
+	return errors.New("not implemented")
+}
+
+// registerRPC for a given topic with an expected protobuf message type.
+func (r *RegularSync) registerRPC(topic string, base proto.Message, h rpcHandler) {
```

Comment:
maybe a better var name, than `h` ?

Reply:
h seems fine to me

Reply:
`h` & `v` are OK for go, but i was on the fence about this anyway. Since it came up in feedback, I'll add a more verbose name for clarity